File: c++/p003.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

int primefactor(int n){
    vector<int> prime_factors;
    int c = 2; // smallest factor
    prime_factors.push_back(c);
    while (n > 1){
        if(n % c == 0){
            cout << c << endl;
            n = n / c;
            prime_factors.push_back(c); // append only prime factors
        }
        else{
            c = c + 1;
        }
    }
    int maxf = *max_element(prime_factors.begin(),prime_factors.end());
    return maxf;
}

int main(int argc, char** argv){
    if (argc < 2){
        cout << "Usage: " << argv[0] << " <number>" << std::endl;
        return 1;
    }
    u_int64_t n = stoi(argv[1]);
    cout << "Project Euler 3" << endl;
    cout << primefactor(n)<<endl;
    return 0;
}

File: python/p003.py
import numpy as np
import sys
"""
What is the largest prime factor of the number
A prime number is a number that:
    Is greater than 1
    Can only be divided evenly by 1 and itself
Prime factors of a number
    prime numbers that, when multiplied together, give you the original number.
"""
def prime_factorization_by_division():
    n = int(sys.argv[1])
    prime_factors = []
    c = 2 # smallest prime number
    prime_factors.append(c)
    while(n > 1):
        if(n % c == 0):
            #print(c, end=" ")
            n = n / c
            prime_factors.append(c) # only keep prime factors
        else:
            c = c + 1
            #prime_factors.append(c) # all factors
    #maxF = max(np.array(prime_factors))
    return list(set(prime_factors))
if __name__ == "__main__":
    print ('Project Euler 3')
    a = prime_factorization_by_division()
    print (a)
    #print (a)
